src/raw/poll_evented.rs

```diff
@@ -210,14 +216,14 @@ where
     ///
     /// The `mask` argument specifies the readiness bits to clear. This may not
     /// include `writable` or `hup`.
-    pub fn clear_read_ready(&self, waker: &Waker) -> io::Result<()> {
+    pub fn clear_read_ready(&self, cx: &mut Context<'_>) -> io::Result<()> {
         self.inner
             .read_readiness
             .fetch_and(!mio::Ready::readable().as_usize(), Relaxed);
 
-        if self.poll_read_ready(waker)?.is_ready() {
+        if Pin::new(self).poll_read_ready(&mut cx)?.is_ready() {
```

Comment:
I think this is supposed to be `cx` not `&mut cx`. Then, `poll_read_ready` needs `Pin<&mut Self>` but this is creating a `Pin<&Self>` which `poll_ready_ready` is not implemented for. It might be possible to handle this with an implementation of `AsyncReadReady` for both `Pin<&mut Self>` and `&self` to provide the necessary functionality. I actually think this method needs to take `Pin<&mut Self>`.